### ISSUE TITLE
Copy allure-report/history in allure-results before generation, for allure trends to work?

### ISSUE THREAD

**Issue by anrock1985** [fescobar/allure-docker-service#28]
Hi, it is possible?

**Comment by fescobar** [comment]
@anrock1985 
Can you try with `beta` version, please?
Enable `KEEP_HISTORY` environment variable like in the example.
```
version: '3'
services:
  allure:
    image: "frankescobar/allure-docker-service:beta"
    environment:
      CHECK_RESULTS_EVERY_SECONDS: 1
      KEEP_HISTORY: "TRUE"
    ports:
      - "4040:4040"
      - "5050:5050"
    volumes:
      - ${PWD}/allure-results:/app/allure-results
```
![image](https://user-images.githubusercontent.com/1236418/61809825-f1847400-ae35-11e9-8e0d-c2eff2568a20.png)
![image](https://user-images.githubusercontent.com/1236418/61809841-f9dcaf00-ae35-11e9-857a-50670128c60c.png)




Let me know if it's working as you expected.

**Comment by fescobar** [comment]
You can use this endpoint to clean the history 
`curl -X GET http://allure:5050/clean-history -ik`

**Comment by anrock1985** [comment]
Thanks for fast reaction =)
Some tests from me:

`@dev26:~/IdeaProjects/ntk-crm-autotests/target$ **docker run -p 80:4040 -p 5050:5050 -e CHECK_RESULTS_EVERY_SECONDS=3 -e KEEP_HISTORY="TRUE" -v ${PWD}/allure-results:/app/allure-results --name allure-docker-service frankescobar/allure-docker-service:beta**
ALLURE_VERSION: 2.12.1
Generating default report
Overriding configuration
Checking Allure Results every 3 second/s
Generating report
Detecting new results...
Generating report
 * Serving Flask app "app" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://0.0.0.0:5050/ (Press CTRL+C to quit)
Report successfully generated to allure-report
Report successfully generated to allure-report
Starting web server...
2019-07-25 01:17:49.308:INFO::main: Logging initialized @280ms to org.eclipse.jetty.util.log.StdErrLog
Can not open browser because this capability is not supported on your platform. You can use the link below to open the report manually.
Server started at <http://172.17.0.2:4040/>. Press <Ctrl+C> to exit


>>>> Generating allure-results from tests here


Detecting new results...
mkdir: cannot create directory ‘/app/allure-results/history’: No such file or directory
Copying history from previous results...
cp: cannot create directory '/app/allure-results/history': No such file or directory
Generating report
Report successfully generated to allure-report
`
Case:
Starting container with prepared allure-results
Just after starting container we have generated report
Executing some tests to update allure-results
Allure container detects updates and starting to copy history folder
"mkdir: cannot create directory ‘/app/allure-results/history’: No such file or directory" **BUG**
Allure shows us a blank report

**Comment by fescobar** [comment]
@anrock1985 let me check and see in detail. I will have news in a few days.

**Comment by fescobar** [comment]
I've found another issue not related to that. 
If you want to mount a volume in docker you need to have a directory created in advance. The same it's not a good idea to have the `allure-results` inside the `target` directory because each time you run a command surely you need to clean first with `mvn clean`, for that reason it doesn't exist that directory. You can put that directory outside the target like this: https://github.com/fescobar/allure-docker-service/tree/master/allure-docker-java-testng-example
If you don't want to do that you should create the `allure-results` directory before starting docker container using any script.

Please, remove your local `beta` image and try again with my suggestions. I've updated the `beta` version.

**Comment by anrock1985** [comment]
removed beta image

moved allure-results to project base dir

dev26:~/IdeaProjects/ntk-crm-autotests$ docker run -p 4040:4040 -p 5050:5050 -e CHECK_RESULTS_EVERY_SECONDS=3 -e KEEP_HISTORY="TRUE" -v ${PWD}/allure-results:/app/allure-results --name allure-docker-service frankescobar/allure-docker-service:beta

**all works like it should, thanks for great work!**

Before that, i tried mvn test (without clean). For some reason, even if all folders in ./target have not been removed, just updated, linux "ls" command could not see folder structure. After "cd ../ & cd target", "ls" was starting to see folder structure again. Its weird

![trends](https://user-images.githubusercontent.com/12438651/61944324-a26e4880-afc7-11e9-8623-726b546869e2.png)

**Comment by fescobar** [comment]
@anrock1985 Thanks to you. In a few mins I will release the version `2.12.1` in there you will the history working. I will remove the `beta` after that.

**Comment by fescobar** [comment]
@anrock1985 Version `2.12.1` was released, don't use `beta`. Documentation is up to date on README.md
